src/udp.c

```diff
@@ -41,6 +43,19 @@ enftun_udp_connect_addr(struct enftun_udp* udp, struct sockaddr* connect_addr)
         return -1;
     }
 
+    if (mark > 0)
+    {
+        if ((rc = setsockopt(udp->fd, SOL_SOCKET, SO_MARK, &mark,
+                             sizeof(mark))) < 0)
+        {
+            enftun_log_error("UDP: Failed to set mark %d: %s\n", mark,
+                             strerror(errno));
+            rc = -errno;
+            close(udp->fd);
+            return rc;
+        }
+    }
+
     rc = connect(udp->fd, connect_addr, sizeof(struct sockaddr));
     if (0 != rc)
     {
```

Comment:
I also didn't notice before, but the `%s` is missing from the following line, so the `strerror(errno)` result doesn't actually appear in the log.  This PR is a good chance to fix that too.  I'm suprised the travis CI builds didn't fail with a warning about an unused argument...